fix: enable prettier pre commit check on jsonc files

stevenh opened the pull request:
Enable the pre commit prettier check on jsonc files by enabling it for all files which identify as text.

Disable always run as passing no files would just error.

Enable color in pre commit hook as it has no impact on action but when run locally allows the user to see the issue more readily.

hyperupcall commented:
The `always_run` change is great. But wondering, doesn't the change from `--write` to `--read` make it so that `pre-commi.ci` no longer automatically formats, commits, and pushes (the formatted code) to the PR? That behavior is important

stevenh commented:
> The `always_run` change is great! But wondering, doesn't the change from `--write` to `--read` make it so that `pre-commi.ci` no longer automatically formats, commits, and pushes (the formatted code) to the PR? That behavior is important (I also worry that Prettier will error when it sees a `text` file that it does not recognize)

When I run it locally it does update the files, did I miss something?
```shell
schemastore$ pre-commit run
prettier.................................................................Failed
- hook id: prettier
- files were modified by this hook

[warn] src/schema-validation.jsonc
[warn] Code style issues fixed in the above file.

codespell................................................................Passed
schemastore$ git st
On branch chore/pre-commit-jsonc
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   src/schema-validation.jsonc

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   src/schema-validation.jsonc
```

stevenh commented:
FYI the reason I changed from `--write` to `--check` is it doesn't tell you which files failed otherwise which found frustrating.

stevenh commented:
Checking why it doesn't error with an unknown file, it default to ignoreUnknown:true:
```shell
pre-commit run -v
prettier.................................................................Passed
- hook id: prettier
- duration: 0.09s

[debug] normalized argv: {"":[".pre-commit-config.yaml","src/something-else.wibble-unknown"],"cache":false,"check":true,"color":false,"editorconfig":true,"ignoreUnknown":true,"write":true,"logLevel":"debug","config":".prettierrc.cjs","ignorePath":[".gitignore"],"configPrecedence":"cli-override","debugRepeat":0,"plugins":[],"_":[".pre-commit-config.yaml","src/something-else.wibble-unknown"],"__raw":{"_":[".pre-commit-config.yaml","src/something-else.wibble-unknown"],"cache":false,"check":true,"color":false,"editorconfig":true,"ignore-unknown":true,"write":true,"log-level":"debug","config":".prettierrc.cjs","ignore-path":".gitignore","config-precedence":"cli-override","debug-repeat":0,"plugin":[]}}
Checking formatting...
[debug] load config file from '.prettierrc.cjs'
[debug] loaded options `{"useTabs":false,"tabWidth":2,"endOfLine":"lf","plugins":["/Users/steve/code/github.com/SchemaStore/schemastore/node_modules/prettier-plugin-sort-json/dist/index.js","/Users/steve/code/github.com/SchemaStore/schemastore/node_modules/prettier-plugin-toml/lib/index.cjs"],"semi":false,"singleQuote":true,"trailingComma":"all","jsonRecursiveSort":true,"jsonSortOrder":"{\"/^[^\\\\d+]/\":\"none\",\"/^\\\\d+/\":\"none\"}"}`
[debug] applied config-precedence (cli-override): {"singleQuote":true,"semi":false,"trailingComma":"all","jsonRecursiveSort":true,"jsonSortOrder":"{\"/^[^\\\\d+]/\":\"none\",\"/^\\\\d+/\":\"none\"}","endOfLine":"lf","plugins":["/Users/steve/code/github.com/SchemaStore/schemastore/node_modules/prettier-plugin-sort-json/dist/index.js","/Users/steve/code/github.com/SchemaStore/schemastore/node_modules/prettier-plugin-toml/lib/index.cjs"],"tabWidth":2,"useTabs":false}
[debug] load config file from '.prettierrc.cjs'
[debug] loaded options `{"endOfLine":"lf","plugins":["/Users/steve/code/github.com/SchemaStore/schemastore/node_modules/prettier-plugin-sort-json/dist/index.js","/Users/steve/code/github.com/SchemaStore/schemastore/node_modules/prettier-plugin-toml/lib/index.cjs"],"semi":false,"singleQuote":true,"trailingComma":"all","jsonRecursiveSort":true,"jsonSortOrder":"{\"/^[^\\\\d+]/\":\"none\",\"/^\\\\d+/\":\"none\"}"}`
[debug] applied config-precedence (cli-override): {"singleQuote":true,"semi":false,"trailingComma":"all","jsonRecursiveSort":true,"jsonSortOrder":"{\"/^[^\\\\d+]/\":\"none\",\"/^\\\\d+/\":\"none\"}","endOfLine":"lf","plugins":["/Users/steve/code/github.com/SchemaStore/schemastore/node_modules/prettier-plugin-sort-json/dist/index.js","/Users/steve/code/github.com/SchemaStore/schemastore/node_modules/prettier-plugin-toml/lib/index.cjs"]}
All matched files use Prettier code style!

codespell................................................................Passed
- hook id: codespell
- duration: 0.05s
- ```

hyperupcall commented:
> When I run it locally it does update the files, did I miss something?

I see! When I was first testing this workflow out in pre-commit CI, I must have missed something else, or possibly assumed/misread that `--check` and `--write` were mutually exclusive. (it should really be called something like `--print`). I recall that the output of `--check` may not have been formatted correctly on pre-commit.ci.

In any case, thanks for these fixes! I don't generally use pre-commit as it's extremely slow on my machine (I use `npm run prettier:fix` instead
